src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,873 @@
+		//	}
+		//}
+
+		//int ViewModel.GetNextIndexInFlatTree(const TreeViewNode& node)
+		//{
+		//	unsigned int index = 0;
+		//	bool isNodeInFlatList = IndexOfNode(node, index);
+
+		//	if (isNodeInFlatList)
+		//	{
+		//		index++;
+		//	}
+		//	else
+		//	{
+		//		// node is Root node, so next index in flat tree is 0
+		//		index = 0;
+		//	}
+		//	return index;
+		//}
+
+		//// When ViewModel receives a event, it only includes the sender(parent TreeViewNode) and index.
+		//// We can't use sender[index] directly because it is already updated/removed
+		//// To find the removed TreeViewNode:
+		////   calcuate allOpenedDescendantsCount in sender[0..index-1] first
+		////   then add offset and finally return TreeViewNode by looking up the flat tree.
+		//TreeViewNode ViewModel.GetRemovedChildTreeViewNodeByIndex(TreeViewNode const& node, unsigned int childIndex)
+		//{
+		//	unsigned int allOpenedDescendantsCount = 0;
+		//	for (unsigned int i = 0; i < childIndex; i++)
+		//	{
+		//		TreeViewNode calcNode = node.Children().GetAt(i).as< TreeViewNode > ();
+		//		if (calcNode.IsExpanded())
+		//		{
+		//			allOpenedDescendantsCount += GetExpandedDescendantCount(calcNode);
+		//		}
+		//	}
+
+		//	unsigned int childIndexInFlatTree = GetNextIndexInFlatTree(node) + childIndex + allOpenedDescendantsCount;
+		//	return GetNodeAt(childIndexInFlatTree);
+		//}
+
+		//int ViewModel.CountDescendants(const TreeViewNode& value)
+		//{
+		//	int descendantCount = 0;
+		//	unsigned int size = value.Children().Size();
+		//	for (unsigned int i = 0; i < size; i++)
+		//	{
+		//		var childNode = value.Children().GetAt(i).as< TreeViewNode > ();
+		//		descendantCount++;
+		//		if (childNode.IsExpanded())
+		//		{
+		//			descendantCount = descendantCount + CountDescendants(childNode);
+		//		}
+		//	}
+
+		//	return descendantCount;
+		//}
+
+		private uint IndexOfNextSibling(TreeViewNode childNode)
+			{
+				//	var parentNode = childNode.Parent();
+				//	unsigned int stopIndex;
+				//	bool isLastRelativeChild = true;
+				//	while (parentNode && isLastRelativeChild)
+				//	{
+				//		unsigned int relativeIndex;
+				//		parentNode.Children().IndexOf(childNode, relativeIndex);
+				//		if (parentNode.Children().Size() - 1 != relativeIndex)
+				//		{
+				//			isLastRelativeChild = false;
+				//		}
+				//		else
+				//		{
+				//			childNode = parentNode;
+				//			parentNode = parentNode.Parent();
+				//		}
+				//	}
+
+				//	if (parentNode)
+				//	{
+				//		unsigned int siblingIndex;
+				//		parentNode.Children().IndexOf(childNode, siblingIndex);
+				//		var siblingNode = parentNode.Children().GetAt(siblingIndex + 1);
+				//		IndexOfNode(siblingNode, stopIndex);
+				//	}
+				//	else
+				//	{
+				//		stopIndex = Size();
+				//	}
+
+				//	return stopIndex;
+				throw new NotImplementedException();
+			}
+
+			private uint GetExpandedDescendantCount(TreeViewNode parentNode)
+			{
+				//	unsigned int allOpenedDescendantsCount = 0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)